File: apps/web/lib/confidence.ts
// Scores are 0..1 but sit low because final = min(structural, visual) *
// commitment, so the tier bands below are deliberately not even thirds.

import type { ConfidenceTier, ConfidenceTierInfo } from "./types";

export function confidenceTier(score: number): ConfidenceTierInfo {
  if (score >= 0.5) return { key: "high", word: "Strong" };
  if (score >= 0.25) return { key: "mid", word: "Moderate" };
  return { key: "low", word: "Low" };
}

export const TIER_FG: Record<ConfidenceTier, string> = {
  low: "var(--color-down)",
  mid: "var(--color-warn)",
  high: "var(--color-up)",
};

File: apps/api/confidence.py
"""Headline-score → confidence tier. Cut-offs mirror `apps/web/lib/confidence.ts`; keep in sync."""

from __future__ import annotations

from typing import NamedTuple


class ConfidenceTier(NamedTuple):
    key: str
    word: str


def confidence_tier(score: float) -> ConfidenceTier:
    # Bands aren't thirds: final = min(structural, visual) × commitment sits low.
    if score >= 0.50:
        return ConfidenceTier("high", "Strong")
    if score >= 0.25:
        return ConfidenceTier("mid", "Moderate")
    return ConfidenceTier("low", "Low")
